### PR TITLE
Extrinsic mean estimator

### LINKED ISSUE #77: Add the extrinsic mean estimator
It's just the mean in the ambient space and point projection. Not a particularly good estimator but it is fast and simple.

### PR DESCRIPTION
**Pull request by mateuszbaran**
I've made a simple implementation of the extrinsic mean estimator. Resolves #77 .

### CODE REVIEW

File: src/statistics.jl [hunk 1]
```diff
@@ -409,6 +419,21 @@ end
     kwargs...,
 )
 
+function Statistics.mean!(
+    M::Manifold,
```
Comment (kellertuer) [inline src/statistics.jl:423]:
```suggestion
    M:: AbstractEmbeddedManifold,
```
I think this only works for embedded manifolds, so we have to take that into account for this type.

Comment (mateuszbaran) [reply src/statistics.jl:423]: I think we also want it to work for wrapped embedded manifolds (when the outermost thing is not `<:AbstractEmbeddedManifold`).

**Comment by mateuszbaran** [comment]
Yes, sure, extrinsic median would also make sense.

File: src/statistics.jl (hunk 1 above)
Comment (kellertuer) [reply src/statistics.jl:423]: should we then maybe restrict it to abstract decorators? Or do you think that is too restrictive?

Comment (mateuszbaran) [reply src/statistics.jl:423]: There doesn't seem to be any good reason for restricting it, the primary reason for restricting argument types should be polymorphic dispatch: https://white.ucc.asn.au/2020/04/19/Julia-Antipatterns.html and here there are no methods we could potentially be concerned about.

**Comment by mateuszbaran** [comment]
Hmm... after thinking about it a bit, what would be the point of using the extrinsic median?

**Comment by kellertuer** [comment]
> Hmm... after thinking about it a bit, what would be the point of using the extrinsic median?

I would ask the same for the mean ;) For both, if possible, I would prefer the intrinsic one, but the extrinsic one might be cheaper.

File: src/statistics.jl (hunk 1 above)
Comment (kellertuer) [reply src/statistics.jl:423]: ok, then let's leave this as is.

**Comment by mateuszbaran** [comment]
The extrinsic mean can be much cheaper than the intrinsic one so there is at least one good reason. I'm not sure how much performance could be gained in the case of median, there doesn't seem to be a fast way to estimate the median in the Euclidean space anyway. Anyway, it looks like something for another PR.

**Comment by kellertuer** [comment]
Sure we can do it in another PR.

The median is still reasonably cheap in the Euclidean space (sorting once basically) compared to manifolds (nonsmooth optimization, so not even a gradient descent applicable).

**Comment by mateuszbaran** [comment]
Sorting only works in R^1, in R^d things are a bit more complicated: https://arxiv.org/abs/1606.05225 .

**Comment by kellertuer** [comment]
Ah, you're right, still the multivariate Euclidean case should still be more efficient than the manifold-case; all those methods look better than doing our current method on manifolds (and I am not aware of a better one in general).

**Comment by mateuszbaran** [comment]
Yes, these methods for R^d are very likely faster. The problem is I don't think any of our current dependencies implements any of these methods for multivariate Euclidean.

**Comment by kellertuer** [comment]
Ah, I warnt aware of that, still; the proximal maps involved on Euclidean space are faster (just a plus) compared to the spherical ones, for example. So conceptually we could still do that,
and add a more clever on by dependency or from Julia Base, when it is available.

So conceptually I would still opt for an extrinsic median, but we can do that in a different PR and then this is fine.

**Comment by mateuszbaran** [comment]
OK, I'll then open an issue for extrinsic median and cover that one last line that I've changed.

**Pull request merged**